### PR TITLE
Implement stt stage-files step

### LINKED ISSUE #1360: `speechToTextWF` `stage-files`: stage transcription files generated by speech_to_text_generation_service
_this is either blocked by #1359, or #1359 should be combined into this ticket_

for now, blocked by the implementation of preceding steps in `speechToTextWF`, but we could probably parallelize this with work on the preceding steps, if we wanted.  we might be able to work off of the `ocrWF` equivalent as an example?

this is the speechToTextWF equivalent of [`ocrWF` `stage-files`](https://github.com/sul-dlss/common-accessioning/blob/main/lib/robots/dor_repo/ocr/stage_files.rb)

stub code in `speechToTextWF`: https://github.com/sul-dlss/common-accessioning/blob/main/lib/robots/dor_repo/speech_to_text/stage_files.rb

files will be in "out" subfolder under `druid_vX` folder in base s3 bucket, which is identifier by the `output_location` method defined here: https://github.com/sul-dlss/common-accessioning/pull/1378/files#diff-1138dcb03554bc6d7579006dc0438967e446fcdb8c45f92e81e3631c087b5b5eR68-R70

part of https://github.com/sul-dlss/common-accessioning/issues/1363

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made? 🤔

Fixes #1360 - download files created by the speech to text service in S3 from the output folder to the local workspace folder

~~HOLD for testing~~

## How was this change tested? 🤨

New specs
On qa

### CODE REVIEW

File: lib/dor/text_extraction/file_fetcher.rb [hunk 1]
```diff
@@ -52,6 +52,8 @@ def write_file_with_retries(filename:, location:, max_tries: 3)
       end
       # rubocop:enable Metrics/MethodLength
 
+      private
+
       # fetch a file from preservation and send to cloud endpoint
```
Comment (peetucket) [inline lib/dor/text_extraction/file_fetcher.rb:57]: more methods can actually be private

File: config/settings.yml [hunk 2]
```diff
@@ -56,6 +56,7 @@ purl_fetcher:
 # aws related options (currently used for the speech-to-text workflow)
 aws:
   region: 'us-west-2'
+  role_arn: 'role-arn'
```
Comment (peetucket) [inline config/settings.yml:59]: some missing template config

File: README.md [hunk 3]
```diff
@@ -58,7 +58,9 @@ Interactive console:
 $ ROBOT_ENVIRONMENT=production ./bin/console
 ```
 
-Running a single robot step manually (without checking current workflow status):
+Running a single robot step manually (without checking current workflow status).  Note the workflow/step name should be the Module::Class name and not the workflow/step name
+(e.g. "Accession::Publish" or "SpeechToText::FetchFiles")
+
```
Comment (peetucket) [inline README.md:63]: improved documentation

File: spec/lib/dor/text_extraction/speech_to_text_spec.rb [hunk 4]
```diff
@@ -22,13 +22,6 @@
   let(:druid) { 'druid:bc123df4567' }
   let(:bare_druid) { 'bc123df4567' }
 
-  def build_file(sdr_preserve, shelve, filename)
-    extension = File.extname(filename)
-    mimetype = { '.m4a' => 'audio/mp4', '.mp4' => 'video/mp4', '.txt' => 'text/plain' }
-    sdr_value = instance_double(Cocina::Models::FileAdministrative, sdrPreserve: sdr_preserve, shelve:)
-    instance_double(Cocina::Models::File, administrative: sdr_value, hasMimeType: mimetype[extension], filename:)
-  end
-
```
Comment (peetucket) [inline spec/lib/dor/text_extraction/speech_to_text_spec.rb:31]: moved to `spec_helper` so it can be used in other tests

**Review by jmartin-sul — APPROVED** [review]

**Pull request merged**